### PR TITLE
Clean up Msgs and Codecs.

### PR DESCRIPTION
**Pull request by SpicyLemon**
## Description

This PR does the following:
* Adds unit tests ensuring the new `signer` stuff works for all our `Msg`s.
* Fixes the signers of a few `Msg`s.
* Removes all the `GetSigners` methods as well as some other ones previously needed by the `Msg` interface.
* Standardizes and cleans up all the `codec.go` files.
* Removes all the `ModuleCdc` variables.
* Uses the same options on the `Coins` proto fields that the SDK does.
* Switch to auto-generated `String` and `Equal` methods for most proto messages.

Related to:
* #1760

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by nullpointer0x00 — APPROVED** [review]

File: proto/cosmos/sanction/v1beta1/sanction.proto [hunk 1]
```diff
@@ -13,15 +14,23 @@ message Params {
   // If this is zero, immediate sanctioning is not available.
   // Otherwise, if a sanction governance proposal is issued with a deposit at least this large, a temporary sanction
   // will be immediately issued that will expire when voting ends on the governance proposal.
-  repeated cosmos.base.v1beta1.Coin immediate_sanction_min_deposit = 1
-      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+  repeated cosmos.base.v1beta1.Coin immediate_sanction_min_deposit = 1 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (amino.dont_omitempty)   = true,
+    (amino.encoding)         = "legacy_coins"
```
Comment (Taztingo) [inline proto/cosmos/sanction/v1beta1/sanction.proto:21]: Why are these needed, and do we even use amino encoding anymore?

File: proto/cosmos/sanction/v1beta1/sanction.proto [hunk 2]
```diff
@@ -13,15 +14,23 @@ message Params {
...
   // If this is zero, immediate unsanctioning is not available.
   // Otherwise, if an unsanction governance proposal is issued with a deposit at least this large, a temporary
   // unsanction will be immediately issued that will expire when voting ends on the governance proposal.
-  repeated cosmos.base.v1beta1.Coin immediate_unsanction_min_deposit = 2
-      [(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+  repeated cosmos.base.v1beta1.Coin immediate_unsanction_min_deposit = 2 [
+    (gogoproto.nullable)     = false,
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (amino.dont_omitempty)   = true,
+    (amino.encoding)         = "legacy_coins"
```
Comment (Taztingo) [inline proto/cosmos/sanction/v1beta1/sanction.proto:32]: Same as above...

File: x/trigger/types/codec.go [hunk 3]
```diff
@@ -1,19 +1,16 @@
...
-// ignoring RegisterLegacyAminoCodec registers all the necessary types and interfaces for the
-// double check
+// RegisterInterfaces registers concrete implementations for this module.
 func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
-	registry.RegisterImplementations(
-		(*sdk.Msg)(nil),
-		&MsgCreateTriggerRequest{},
-		&MsgDestroyTriggerRequest{},
-	)
+	messages := make([]proto.Message, len(AllRequestMsgs))
```
Comment (Taztingo) [inline x/trigger/types/codec.go:11]: Do we need an extra copy/heap allocation for this? RegisterImplementations is only reading, and this would also allow us to get rid of the new import.

`registry.RegisterImplementations((*sdk.Msg)(nil), AllRequestMsgs...)`

**Review by Taztingo — APPROVED** [review]

File: proto/cosmos/sanction/v1beta1/sanction.proto (hunk 1 above)
Comment (SpicyLemon) [reply proto/cosmos/sanction/v1beta1/sanction.proto:21]: I'm not entirely sure. I added them here because the SDK added them to all their `Coins` fields, but I couldn't find anything that indicated why they were added or what exactly they do.  The `amino.proto` documentation makes it sound like it's related to JSON encoding, though.

I took these lines out and regenerated stuff, and the only thing that changed was the `fileDescriptor_...` byte slice which is pretty much just the contents of that file. But that's also the case with options like `(cosmos_proto.scalar) = "cosmos.AddressString"` or `(cosmos.msg.v1.signer)`.

I did some testing on two versions of a proto message, one with these options, one without. I generated JSON from each in three different ways:`encoding/json.Marshal`, `"github.com/cosmos/cosmos-sdk/codec".Codec#MarshalJSON` and `"github.com/cosmos/cosmos-sdk/codec".LegacyAmino#MarshalJSON`. I used a fully set-up encoding config for the latter two. All six combos produced the same json.

So either:
1. We don't need these options.
2. Or we're not wiring things up right, causing these to not have meaning.
3. Or they affect something else.

File: x/trigger/types/codec.go (hunk 3 above)
Comment (SpicyLemon) [reply x/trigger/types/codec.go:11]: Yeah.  `RegisterImplementations` takes in a `[]proto.Message`, but the `AllRequestMsgs` variable is a `[]sdk.Msg`. Since `proto.Message` is a subset of `sdk.Msg`, Golang can automatically cast a single entry, but it can't cast slices like that, so we have to copy them.

In previous versions of the SDK, the `sdk.Msg` interface was a `proto.Message` with several other methods. It was important to have some sort of type assertion on them to make sure all required methods were defined. That's why the `AllRequestMsgs` variables are all `[]sdk.Msg`; having them in that list gave us the desired type assertion for free.

Over the various versions, though, the SDK has removed methods from the `sdk.Msg` interface, and it is now just an alternative name for `proto.Message`. If we changed `AllRequestMsgs` to a `[]proto.Message`, we wouldn't have to do this copying. I'm not quite sure what other ramifications that might have, though. But I'd rather leave them as `[]sdk.Msg` since that's the type we're registering them as in these codecs.

Comment (Taztingo) [reply x/trigger/types/codec.go:11]: I actually got this mixed up with `attribute/types/codec.go` where `allRequestMsgs` already has the []proto.Message type.
